src/app/public/src/modules/edit-button/edit-button.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { StacheJsonDataService } from '../shared';
+import { StacheNavService } from '../nav';
+
+@Component({
+  selector: 'stache-edit-button',
+  templateUrl: './edit-button.component.html',
+  styleUrls: ['./edit-button.component.scss']
+})
+export class StacheEditButtonComponent implements OnInit {
+  @Input()
+  public editButtonText = 'Edit';
+
+  private url: string;
+  private routeFragment: string;
+
+  public constructor (
+    private dataService: StacheJsonDataService,
+    private navService: StacheNavService,
+    private router: Router
```

Comment:
Any reason we are pulling in the router instead of the StacheRouteService?
https://github.com/blackbaud/stache2/blob/master/src/app/public/src/modules/shared/route.service.ts#L9

Reply:
I didn't know that existed. :P That's why. I'll use that instead.